target/ppc/translate/vsx-impl.c.inc

```diff
@@ -1546,6 +1546,42 @@ static bool trans_LXVKQ(DisasContext *ctx, arg_X_uim5 *a)
     return true;
 }
 
+static bool trans_XVTLSBB(DisasContext *ctx, arg_X_bf_xb *a)
+{
+    TCGv_i64 xb, all_true, all_false, crbf;
+
+    REQUIRE_INSNS_FLAGS2(ctx, ISA310);
+    REQUIRE_VSX(ctx);
+
+    xb = tcg_temp_new_i64();
+    all_true = tcg_const_i64(1);
+    all_false = tcg_const_i64(1);
+    crbf = tcg_temp_new_i64();
+
+    for (int dw = 0; dw < 2; dw++) {
+        get_cpu_vsr(xb, a->xb, dw);
+        for (int nbyte = 0; nbyte < 8; nbyte++) {
+            tcg_gen_and_i64(all_true, all_true, xb);
+            tcg_gen_andc_i64(all_false, all_false, xb);
+            tcg_gen_shri_i64(xb, xb, 8);
+        }
+    }
+
+    tcg_gen_mov_i64(crbf, all_true);
+    tcg_gen_shli_i64(crbf, crbf, 2);
+    tcg_gen_or_i64(crbf, crbf, all_false);
+    tcg_gen_shli_i64(crbf, crbf, 1);
+
+    tcg_gen_trunc_tl_i32(cpu_crf[a->bf], crbf);
```

Comment:
I believe cpu_crf is always 32 bits long  (TCGv_i32), so you can build it directly from 32-bit registers, without needing to truncate here

Reply:
Could you give me an idea on how to do it the way you proposed? It is always necessary to operate on at least one TCGv_i64 (xb), so there will always be one truncate, right?

Reply:
What I was thinking wouldn't work. I'll send another comment with the revised code, though.

Reply:
BTW, in the revised code I replaced ``tcg_gen_trunc_tl_i32`` with ``tcg_gen_extrl_i64_i32``, because ``cpu_crf[a->bf]`` is always 32-bit and ``crbf`` (or ``temp`` in the revised code) is always 64-bit.

``tcg_gen_trunc_tl_i32`` would be wrong because, on 32 bit hosts, it's defined as ``tcg_gen_mov_i32`` (see include/tcg/tcg-op.h), which expects both arguments to be TCGv_i32.